src/core_functions/scalar/list/list_cosine_similarity.cpp

```diff
@@ -41,6 +45,10 @@ static void ListCosineSimilarity(DataChunk &args, ExpressionState &, Vector &res
 
 		    auto dimensions = left.length;
 
+		    if (dimensions == 0) {
+			    return NullValue<NUMERIC_TYPE>();
```

Comment:
This now returns `NaN`. Can we rework this to throw an error instead? (e.g. `InvalidInputException("The cosine similarity for empty vectors is not defined")`).

Reply:
updated, plz check again